cpp/src/utilities/cuda_memcpy.cu

```diff
@@ -26,15 +28,21 @@ namespace cudf::detail {
 
 namespace {
 
+__global__ void copy_kernel(char const* src, char* dst, size_t n)
```

Comment:
Can you add docs explaining why do we need this kernel, instead of `cudaMemcpyAsync`?